xapian-letor/tests/api_letor.cc

```diff
@@ -263,6 +263,36 @@ DEFINE_TESTCASE(preparetrainingfileonedb, generated)
     return true;
 }
 
+DEFINE_TESTCASE(test_parse_querystring, generated)
+{
+    // All those cases which are not valid.
+    string db_path = get_database_path("apitest_listnet_ranker1",
+				       db_index_one_document);
+    string data_directory = test_driver::get_srcdir() + "/testdata/";
+    string query = data_directory + "parse_query_noopenquote.txt";
+    string qrel = data_directory + "qrelone.txt";
+    string training_data = data_directory + "training_output.txt";
```

Comment:
This variable isn't used.

Reply:
Also, output files should be written to the build tree not the source tree (which are often the same but don't have to be) so `"training_output.txt"` as actually used below is good, and the value computed here wouldn't be.

When building with `srcdir` != `builddir` (also known as a "VPATH build" after the VPATH feature in make used to implement it) one `srcdir` may be shared between several build trees and so generally should be treated as read-only.

Reply:
fixed in a fixup commit.